waymo_raw_data_to_argoverse.py

```diff
@@ -30,205 +40,398 @@
-	check_mkdir(str(Path(json_fpath).parent))
-	save_json_dict(json_fpath, pose_dict)
-
-
-
-if __name__ == '__main__':
-
-	save_images = True
-	save_poses = True
-	save_calibration = True
-
-	main(save_images, save_poses, save_calibration)
+def save_json_dict(
+    json_fpath: Union[str, "os.PathLike[str]"], dictionary: Dict[Any, Any]
+) -> None:
+    """Save a Python dictionary to a JSON file.
 
+    Args:
+    json_fpath: Path to file to create.
+    dictionary: Python dictionary to be serialized.
+    """
+    with open(json_fpath, "w") as f:
+        json.dump(dictionary, f)
 
 
+def get_log_id_from_files(record_dir: str) -> List[str]:
+    """Get the log IDs of the Waymo records from the directory where they are stored
+
+    Args:
+        record_dir: The path to the directory where the Waymo data is stored
+    Returns:
+        log_ids: A list of log IDs from the Waymo dataset
+    """
+    files = glob.glob(f"{record_dir}/*.tfrecord")
+    log_ids = []
+    for file in files:
+        file = file.strip(record_dir)
+        file = file.strip("segment-")
+        file = file.strip("with_camera_labels.tfrecord")
+        log_ids.append(file)
+    return log_ids
+
+
+def main(args: argparse.Namespace) -> None:
+    """ """
+    TFRECORD_DIR = args.waymo_dir
+    ARGO_WRITE_DIR = args.argo_dir
+    track_id_dict = {}
+    img_count = 0
+    log_ids = get_log_id_from_files(TFRECORD_DIR)[:1]
+    for log_id in log_ids:
+        print(log_id)
+        tfrecord_name = f"segment-{log_id}_with_camera_labels.tfrecord"
+        tf_fpath = f"{TFRECORD_DIR}/{tfrecord_name}"
+        dataset = tf.data.TFRecordDataset(tf_fpath, compression_type="")
+        log_calib_json = None
+        for data in dataset:
+            frame = open_dataset.Frame()
+            frame.ParseFromString(bytearray(data.numpy()))
+            # Checking if we extracted the correct log ID
+            assert log_id == frame.context.name
+            # Frame start time, which is the timestamp of the first top lidar spin
+            # within this frame, in microseconds
+            timestamp_ms = frame.timestamp_micros
+            timestamp_ns = int(timestamp_ms * 1000)  # to nanoseconds
+            SE3_flattened = np.array(frame.pose.transform)
+            city_SE3_egovehicle = SE3_flattened.reshape(4, 4)
+            if args.save_poses:
+                dump_pose(city_SE3_egovehicle, timestamp_ns, log_id, ARGO_WRITE_DIR)
+            # Reading lidar data and saving it in point cloud format
+            # We are only using the first range image by default (Waymo provides two range images)
+            # If you want to use the second range image, you can change it in the arguments
+            (
+                range_images,
+                camera_projections,
+                range_image_top_pose,
+            ) = frame_utils.parse_range_image_and_camera_projection(frame)
+            if args.range_image == 1:
+                (
```

Comment:
i would prefer without the parentheses on the return args, so the following instead:

````python
points_ri, cp_points_ri = frame_utils.convert_range_image_to_point_cloud(
    frame, range_images, camera_projections, range_image_top_pose
)
```

Reply:
Sorry `black` was doing this. Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/3b5f39632d27d0d05165fd850e73bd772ceaaaf5)